tkg/client/cluster.go

```diff
@@ -160,7 +160,13 @@ func (c *TkgClient) CreateCluster(options *CreateClusterOptions, waitForCluster
 			return false, errors.Wrap(err, "unable to get cluster configuration")
 		}
 
-		if !c.IsFeatureActivated(constants.FeatureFlagAllowLegacyCluster) {
+		allowLegacyCluster, err := c.TKGConfigReaderWriter().Get(constants.ConfigVariableAllowLegacyCluster)
+		if err != nil {
+			log.Error(err, "ALLOW_LEGACY_CLUSTER variable is set failed")
+			return false, err
+		}
+
+		if allowLegacyCluster != "true" {
```

Comment:
**suggestion:** It is best not to make an already giant method bigger. Instead, create a function encapsulating the added logic. 

The code that you are replacing is simpler. It would be great to preserve that simplicity.

It will also be easier to test and understand.